static/app/components/autoComplete.tsx

```diff
@@ -157,14 +179,13 @@ class AutoComplete<T extends Item> extends React.Component<Props<T>, State<T>> {
    */
   resetHighlightState() {
     // reset items and expect `getInputProps` in child to give us a list of new items
-    this.setState({
-      highlightedIndex: this.props.defaultHighlightedIndex || 0,
-    });
+    this.setState({highlightedIndex: this.props.defaultHighlightedIndex ?? 0});
   }
 
-  handleInputChange =
-    <E extends HTMLInputElement>({onChange}: Pick<GetInputArgs<E>, 'onChange'>) =>
-    (e: React.ChangeEvent<E>) => {
+  makeHandleInputChange<E extends HTMLInputElement>(
```

Comment:
These are all factory functions, renamed them as such.

They also don't need to be instance properties, they can just be class methods